tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator/Main.java

```diff
@@ -60,8 +64,30 @@ public static void main(String[] args) {
             parseFiles(getTracefiles(flags, filesInCoverageDir), LcovParser::parse),
             parseFiles(getGcovInfoFiles(filesInCoverageDir), GcovParser::parse));
 
+    File profdataFile = getProfdataFileOrNull(filesInCoverageDir);
     if (coverage.isEmpty()) {
-      logger.log(Level.SEVERE, "There was no coverage found.");
+      int exitStatus = 0;
+      if (profdataFile == null) {
+        logger.log(Level.SEVERE, "There was no coverage found.");
+        exitStatus = 1;
+      } else {
+        // Coverage generated one profdata report. Bazel doesn't support yet converting profdata
+        // files to lcov. We still want to output a coverage report so we copy the content of
+        // the profdata file to the output file. This is not ideal but it unblocks some Bazel C++
+        // coverage users.
+        // TODO(#5881): Add support for profdata files.
+        logger.log(Level.INFO, "One profdata file was found. Skipping converting to lcov.");
```

Comment:
You can drop the first sentence because it is reported from getProfdataFileOrNull()

Reply:
Oh, my bad, I was referring to line 80: logger.log(Level.INFO, "One profdata file was found. Skipping converting to lcov.");
You already report that one profdata file was found in  getProfdataFileOrNull()